Add Nix flake for reproducible development environment

mzonski opened the pull request:
This PR introduces a Nix flake that creates a consistent development environment with all required system dependencies (wireguard, openresolv) while leveraging familiar Python tooling for package management. Using a flake eliminates "it works on my machine" problems by ensuring everyone has identical development environments regardless of their host OS.

kchiem commented:
I don't use nix or direnv, so I'll have to take your word for these, but I'm guessing .envrc doesn't need the shebang.

mzonski commented:
Got it, removed direnv and cleaned branch. The config gets generated properly
![pia](https://github.com/user-attachments/assets/2e549dae-6cc9-49ae-ab7b-6b51f76a8533)
